core/core.go

```diff
@@ -565,3 +565,22 @@ func getData[T any](api *DefaultAPI, token string, url string) (T, error) {
 
 	return data, nil
 }
+
+const maxBytesLimit int64 = 1024 * 1024 * 10 // 10MB
+
+// re-implementation of io.ReadAll with max limit
+func MaxBytesReadAll(r io.Reader) ([]byte, error) {
+	limitedReader := &io.LimitedReader{
```

Comment:
There is also `io.LimitReader(r io.Reader, n int64)` which wraps reader in `LimitedReader` on it's own